core-library/src/main/java/org/silverpeas/core/index/search/model/SearchResult.java

```diff
@@ -0,0 +1,215 @@
+
+import org.silverpeas.core.contribution.contentcontainer.content.GlobalSilverContent;
+import org.silverpeas.core.i18n.AbstractI18NBean;
+import org.silverpeas.core.util.StringUtil;
+import org.silverpeas.core.util.file.FileServerUtils;
+
+import java.time.LocalDate;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * @author Nicolas Eysseric
+ */
+public class SearchResult extends AbstractI18NBean {
+
+  private String keywords;
+  private LocalDate creationDate;
+  private String creatorId;
+  private LocalDate lastUpdateDate;
+  private String lastUpdaterId;
+
+  private String id;
+  private String componentId;
+  private String type;
+
+  private float score;
+  private String serverName;
+  private String thumbnailURL;
+  private String attachmentFilename;
+  private boolean externalResult = false;
+  private boolean alias = false;
+
+  /**
+   * list of XML form fields used to sort results
+   */
+  private Map<String, String> sortableXMLFormFields = null;
+  private Map<String, String> formFieldsForFacets;
+
+  public static SearchResult fromIndexEntry(MatchingIndexEntry mie) {
+    return new SearchResult(mie);
+  }
+
+  public static SearchResult fromGlobalSilverContent(GlobalSilverContent gsc) {
+    return new SearchResult(gsc);
+  }
+
+  private SearchResult(MatchingIndexEntry mie) {
+    setName(mie.getTitle());
+    setDescription(mie.getPreView());
+    this.keywords = mie.getKeywords(null);
+
+    this.creationDate = getLocalDate(mie.getCreationDate());
+    this.lastUpdateDate = getLocalDate(mie.getLastModificationDate());
+
+    this.creatorId = mie.getCreationUser();
+    this.lastUpdaterId = mie.getLastModificationUser();
+
+    this.id = mie.getObjectId();
+    this.componentId = mie.getComponent();
+    this.type = mie.getObjectType();
+
+    this.score = mie.getScore();
+    this.serverName = mie.getServerName();
+    attachmentFilename = mie.getFilename();
+    externalResult = mie.isExternalResult();
+    alias = mie.isAlias();
+
+    // add sortable fields from XML form
+    sortableXMLFormFields = mie.getSortableXMLFormFields();
+    formFieldsForFacets = mie.getXMLFormFieldsForFacets();
+
+    Iterator<String> languages = mie.getLanguages();
+    while (languages.hasNext()) {
+      String language = languages.next();
+      SearchResultTranslation translation =
+          new SearchResultTranslation(language, mie.getTitle(language), mie.getPreview(language));
+      addTranslation(translation);
+    }
+
+    if (mie.getThumbnail() != null) {
+      if (mie.getThumbnail().startsWith("/")) {
+        // case of a thumbnail picked up in a gallery
+        thumbnailURL = mie.getThumbnail();
+      } else {
+        // case of an uploaded image
+        thumbnailURL = FileServerUtils.getUrl(mie.getComponent(),
+            mie.getThumbnail(), mie.getThumbnailMimeType(), mie.getThumbnailDirectory());
+      }
+    }
+  }
+
+  private SearchResult(GlobalSilverContent gsc) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1213)